File: backend/src/models/Complaint.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Complaint = sequelize.define('Complaint', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  category: {
    type: DataTypes.ENUM(
      'Harassment',
      'Theft',
      'Property Loss',
      'Suspicious Activity',
      'Fraud',
      'Cyber Issue',
      'Other'
    ),
    allowNull: false,
  },
  title: {
    type: DataTypes.STRING(200),
    allowNull: false,
    defaultValue: 'Untitled Complaint',
  },
  description: {
    type: DataTypes.TEXT,
    allowNull: false,
  },
  status: {
    type: DataTypes.ENUM('Received', 'In Progress', 'Resolved'),
    defaultValue: 'Received',
  },
  judgement_details: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
  summary: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
  summarized_at: {
    type: DataTypes.DATE,
    allowNull: true,
  },
  summary_source_hash: {
    type: DataTypes.STRING(64),
    allowNull: true,
  },
  created_by: {
    type: DataTypes.UUID,
    allowNull: false,
    references: { model: 'users', key: 'id' },
  },
}, {
  tableName: 'complaints',
  timestamps: true,
  createdAt: 'created_at',
  updatedAt: 'updated_at',
});

module.exports = Complaint;

File: Frontend/models/complaint.dart
class Complaint {
  final String id;
  final String category;
  final String title;
  final String description;
  final String status;
  final String? judgementDetails;
  final String? summary;
  final String? summarizedAt;
  final String createdBy;
  final String createdAt;
  final String updatedAt;
  final List<Map<String, dynamic>> complainants;
  final List<Map<String, dynamic>> accusedPersons;
  final List<Map<String, dynamic>> media;
  final List<Map<String, dynamic>> statusLog;
  final Map<String, dynamic>? creator;

  Complaint({
    required this.id,
    required this.category,
    required this.title,
    required this.description,
    required this.status,
    this.judgementDetails,
    this.summary,
    this.summarizedAt,
    required this.createdBy,
    required this.createdAt,
    required this.updatedAt,
    this.complainants = const [],
    this.accusedPersons = const [],
    this.media = const [],
    this.statusLog = const [],
    this.creator,
  });

  factory Complaint.fromJson(Map<String, dynamic> json) {
    return Complaint(
      id: json['id'],
      category: json['category'] ?? '',
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      status: json['status'] ?? 'Received',
      judgementDetails: json['judgement_details'],
      summary: json['summary'],
      summarizedAt: json['summarized_at'],
      createdBy: json['created_by'] ?? '',
      createdAt: json['created_at'] ?? '',
      updatedAt: json['updated_at'] ?? '',
      complainants: List<Map<String, dynamic>>.from(json['complainants'] ?? []),
      accusedPersons: List<Map<String, dynamic>>.from(
        json['accusedPersons'] ?? [],
      ),
      media: List<Map<String, dynamic>>.from(json['media'] ?? []),
      statusLog: List<Map<String, dynamic>>.from(json['statusLog'] ?? []),
      creator: json['creator'],
    );
  }
}
